File: assignment 4/posnegzero.cpp
#include<iostream>
using namespace std;
class Difference
{
    int arr[10],i,n,positive=0,negative=0,zero=0;
    public :void accept()
    {
        cout<<"Enter the size of the array: ";
        cin>>n;
        cout<<"Enter the elements in the array: ";
        for(i=0;i<n;i++)
        {
            cin>>arr[i];
        }
    }

    void display()
    {
        for(i=0;i<n;i++)
            if(arr[i]>0)
            {
                cout<<"Positive values are: "<<positive<<endl;
                positive++;
            }
                else if(arr[i]<0)
                 {
                   cout<<"Negative values are: "<<negative<<endl;
                   negative++;
                 }
                    else
                     {
                       cout<<"Zero values are: "<<zero<<endl;
                       zero++;

                     }
                 }

            };

int main()
{
    Difference obj;
    obj.accept();
    obj.display();
}

File: Java assignment(3-4)/Java assignment 4/Pos_Neg_Zero.java
import java.util.*;
public class Pos_Neg_Zero {
	int n,pos=0,neg=0,zero=0;
	int arr[]=new int[100];
	void getDetails()
	{
		Scanner sc=new Scanner(System.in);
		System.out.println("Enter the number of elements: ");
		n=sc.nextInt();
		System.out.println("Enter the elements: ");
		for(int i=0;i<n;i++)
		{
		 arr[i]=sc.nextInt();
		}

		for(int i=0;i<n;i++)
		{
			if(arr[i]>0)
			{
				pos++;
			
			}
			else if(arr[i]<0)
			{
				neg++;
			}
			else if(arr[i]==0)
			{
				zero++;
				
			}
			else
			{
				System.out.println("Enter a valid set of integers");
				
			}
		}
		System.out.println("Positive: "+pos);
		System.out.println("Negative: "+neg);
		System.out.println("Zero: "+zero);
	}
	

	public static void main(String[] args) {
		
Pos_Neg_Zero obj=new Pos_Neg_Zero();
obj.getDetails();
	}

}
